src/meta/src/dashboard/mod.rs

```diff
@@ -206,11 +216,93 @@ pub(super) mod handlers {
 
         Ok(result.into())
     }
+
+    pub async fn heap_profile(
+        Path(worker_id): Path<WorkerId>,
+        Extension(srv): Extension<Service>,
+    ) -> Result<Json<HeapProfilingResponse>> {
+        let worker_node = srv
+            .cluster_manager
+            .get_worker_by_id(worker_id)
+            .await
+            .context("worker node not found")
+            .map_err(err)?
+            .worker_node;
+
+        let client = srv.compute_clients.get(&worker_node).await.map_err(err)?;
+
+        let result = client.heap_profile("".to_string()).await.map_err(err)?;
+
+        Ok(result.into())
+    }
+
+    pub async fn list_heap_profile(
+        Path(worker_id): Path<WorkerId>,
+        Extension(srv): Extension<Service>,
+    ) -> Result<Json<ListHeapProfilingResponse>> {
+        let worker_node = srv
+            .cluster_manager
+            .get_worker_by_id(worker_id)
+            .await
+            .context("worker node not found")
+            .map_err(err)?
+            .worker_node;
+
+        let client = srv.compute_clients.get(&worker_node).await.map_err(err)?;
+
+        let result = client.list_heap_profile().await.map_err(err)?;
+        Ok(result.into())
+    }
+
+    pub async fn analyze_heap(
+        Path((worker_id, file_path)): Path<(WorkerId, String)>,
+        Extension(srv): Extension<Service>,
+    ) -> Result<Response> {
+        if srv.ui_path.is_none() {
+            bail!("Should provide ui_path");
+        }
```

Comment:
Can you elaborate on why we need this?

Reply:
Probably just sth forget to delete after some refactoring.